webapp/src/ts/services/select2-search.service.ts

```diff
@@ -10,21 +10,26 @@ import { SessionService } from '@mm-services/session.service';
 import { SettingsService } from '@mm-services/settings.service';
 import { ContactMutedService } from '@mm-services/contact-muted.service';
 import { TranslateService } from '@mm-services/translate.service';
+import { SearchTelemetryService } from '@mm-services/search-telemetry.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class Select2SearchService {
+  private selectEl;
+  private currentQuery: string | null = null;
+  private onContactSelect;
```

Comment:
I don't think it is going to work to have this state here at the service level.  I double-checked and confirmed that if you have a form with more than one `select-contact` field, then the `init` function below is going to get called for each of the contact selectors in the form.  (And this state will only be set for the last one).

Maybe I am missing something here, but is there a reason we cannot setup the `select2:select` event listener somewhere like the `initSelect2` method (for each given selector) instead of doing it in the generic `query` method?

Reply:
That's a good catch... I landed with registering this event listener in the `query` method because this is where I had access to the user's search to hook up with the contact the user would end up selecting.
I also couldn't find anything in [select2 docs](https://select2.org/) to retrieve the search text programmatically without writing code that looks like a pile of hacks with a mix of jQuery and DOM element querying :sweat_smile: 

But! I think I found [a solution](https://stackoverflow.com/a/44678950). I'll come back and resolve this convo if this works.